Worked on issues

amitpercept86 opened the pull request:
1. #1 Replaced all $_GET with laravel request helper function.
2. #2 Removed token refresh logic from boot method of package, Instead using schedule task now.
3. #3 I have added settings in .gitattributes and replaced all existing \r\n to \n. I am not sure whether this will work or not. If it's not working please advice what I can do here in my VS code running on Windows system.
4. #4 Done. You can controll this function using environment variable PERCEPT_DROPBOX_FILES_SORT_BY and PERCEPT_DROPBOX_FILES_SORT_ORDER. I have mention all possible values for this variable in readme file.
5. #5 Done. Plugin has install command which publish the configuration. Please check this in readme file.
6. #6 Done.

amitpercept86 committed:
worked on issues

amitpercept86 committed:
Update README.md

Added line break in readme file

willwoodlief commented:
Three issues, these can be tested and fixed by installing this on a fresh install of a skeleton:

### Cannot install
Still cannot install this because the database migration has not been run (cannot be run) before this is installed, but the code in public function packageBooted() is trying to use it during the install process. 

The cron task is supposed to refresh and setup the token to be used.
So, you will need to figure out how to make sure the token is working and refreshed after the package is installed. Maybe one solution is to set up a run one time event in the cron system set for a minute after the install. Probably better solutions ...

### No publishing the config
Remove the ->hasInstallCommand(function (InstallCommand $command)  completely. We do not want to change any files in the main project, so we cannot publish anything


### Readme is inaccurate on composer command

The readme is wrong in the directions how to install this with composer. You just want to run the
 
    `composer update kunalvarma05/dropbox-php-sdk`

willwoodlief commented:
Leaving this pull request open so new commits and comments can be made here

amitpercept86 committed:
Fixed table not exists issue while installation

amitpercept86 commented:
1. You can install plugin now. I am checking table exists prior to accessing it. So you will not get error while installation.
I did research whether we can set one time cron job after plugin installed, but I don't think that can help here. I think we must have to assign a token to Dropbox singleton on every request, so we just can't move/remove code that assign token to Dropbox instance inside packageBooted. 
Further, I have removed create table migration file because now I am creating table at runtime if it's not exists using Schema facade.   

2. Removed installation command. We don't have any publishable things now.

3. Changed readme and now using `composer update kunalvarma05/dropbox-php-sdk`

willwoodlief commented:
Hi,

We really need that migration file used again, could you please have all db changes done via that?  Its important to be able to undo the migration, and migrations make things much easier

amitpercept86 committed:
Added migration file back and changed readme and other files

amitpercept86 commented:
I have added DB migration back and did required changes in other files. I have tested it on fresh installation of  https://github.com/willwoodlief/scm-powered-skeleton

willwoodlief commented:
I can now install it and it works! Great!!! One final change? You have the dropbox folder hard coded to documents, in two places. Possible to make this folder configerable in the .env file?

amitpercept86 committed:
Added upload_folder config and validated token on connect route

amitpercept86 committed:
Changed readme

amitpercept86 commented:
As requested, I have added configurable variable for setting upload directory for Dropbox file upload.

willwoodlief commented:
That works ok
